lib/doorkeeper/grape/helpers.rb

```diff
@@ -0,0 +1,59 @@
+module Doorkeeper
+  module Grape
+    module Helpers
+      extend ActiveSupport::Concern
+
+      included do
+        helpers do
+          def doorkeeper_token
+            @_doorkeeper_token ||= OAuth::Token.authenticate request, *Doorkeeper.configuration.access_token_methods
+          end
+
+          def valid_doorkeeper_token?(*scopes)
+            doorkeeper_token && doorkeeper_token.acceptable?(scopes)
+          end
+
+          # endpoint specific scopes > parameter scopes > default scopes
+          def doorkeeper_authorize!(*scopes)
+            endpoint_scopes = env['api.endpoint'].options[:route_options][:scopes]
+            scope_string = Doorkeeper::OAuth::Scopes.from_array(scopes).all.inspect if scopes && !scopes.empty?
+            scope_string = Doorkeeper::OAuth::Scopes.from_array(endpoint_scopes).all.inspect if endpoint_scopes
+            scopes = Array.wrap(scope_string) if scope_string
+            scopes = Doorkeeper.configuration.default_scopes if scopes.empty?
+
+            unless valid_doorkeeper_token?(*scopes)
+              if !doorkeeper_token || !doorkeeper_token.accessible?
+                error = OAuth::InvalidTokenResponse.from_access_token(doorkeeper_token)
+                options = doorkeeper_unauthorized_render_options
+              else
+                error = OAuth::ForbiddenTokenResponse.from_scopes(scopes)
+                options = doorkeeper_forbidden_render_options
+              end
+              headers.merge!(error.headers.reject { |k| ['Content-Type'].include? k })
+              doorkeeper_error_renderer(error, options)
+            end
+          end
+
+          def doorkeeper_unauthorized_render_options
+            nil
+          end
+
+          def doorkeeper_forbidden_render_options
+            nil
+          end
+
+          def doorkeeper_error_renderer(error, options = {})
```

Comment:
Unused method argument - `options`. If it's necessary, use `_` or `_options` as an argument name to indicate that it won't be used.